dcm4che-net/src/main/java/org/dcm4che3/net/service/QueryRetrieveLevel.java

```diff
@@ -96,6 +108,43 @@ public void validateRetrieveKeys(Attributes attrs,
         check(attrs.validate(retrieveKeysIOD(rootLevel, relational)));
     }
 
+    public static QueryRetrieveLevel validateRetrieveIdentifier(Attributes keys, Set<QueryRetrieveLevel> qrLevels,
+            boolean relational, boolean lenient) throws DicomServiceException {
+        return validateIdentifier(keys, qrLevels, relational, lenient);
+    }
+
+    private static QueryRetrieveLevel validateIdentifier(Attributes keys, Set<QueryRetrieveLevel> qrLevels,
+             boolean relational, boolean lenient) throws DicomServiceException {
```

Comment:
Why did you use a private method? It doesn't appear to be used anywhere else so it could be the body of the validateRetrieveIdentifier method

Reply:
It's copy/paste from https://github.com/dcm4che/dcm4che/blob/master/dcm4che-net/src/main/java/org/dcm4che3/net/service/QueryRetrieveLevel2.java.

Gunter used this structure to allow several overloads to use the same method. However, I agree this could be done in one public method.